### ISSUE TITLE
akka.cluster.MemberStatus ought to be a sealed type

### ISSUE THREAD

**Issue by retronym** [akka/akka#19679]
I recently improved exhaustivity warnings in the Scala 2.12 compiler. Running this against Akka sources, we get this warning:

```
[akka] [warn] /home/jenkins/workspace/scala-2.12.x-integrate-community-build/target-0.9.5/project-builds/akka-622a540c67e9bc2514e3685a0f16c7ad317992b0/akka-cluster/src/main/scala/akka/cluster/Member.scala:142: match may not be exhaustive.
[akka] [warn] It would fail on the following inputs: (Up, _), (_, Up), (_, _)
[akka] [warn]   def highestPriorityOf(m1: Member, m2: Member): Member = (m1.status, m2.status) match {
[akka] [warn] 
```

This will go away if we seal `MemberStatus`.

**Comment by johanandren** [comment]
Sounds reasonable to me. Breaks source compatibility but I cannot see how anyone would have provided their own subtype of `MemberStatus`, must be an oversight on our part.

**Comment by 2m** [comment]
Fixed in #19680.
